### PR TITLE
ci(workflow): add cache to workflows using actions/setup-node

### PR DESCRIPTION
**Pull request by oscard0m**
## Description

Add `cache` to workflows using `actions/setup-node`

## Context

`setup-node` GitHub Action just released a new option to add cache to steps using it.

You can find the details here: https://github.blog/changelog/2021-07-02-github-actions-setup-node-now-supports-dependency-caching/

---

🤖 This PR has been generated automatically by [this octoherd script](https://github.com/oscard0m/octoherd-script-add-cache-to-node-github-action), feel free to run it in your GitHub user/org repositories! 💪🏾

### CODE REVIEW

File: .github/workflows/CI.yml [hunk 1]
```diff
@@ -7,9 +7,10 @@ jobs:
 
     steps:
       - uses: actions/checkout@v1
-      - uses: actions/setup-node@v1
+      - uses: actions/setup-node@v2
         with:
           node-version: "12.x"
+          cache: npm
```
Comment (oscard0m) [inline .github/workflows/CI.yml:13]:
```suggestion
          cache: yarn
```

File: .github/workflows/CI.yml [hunk 2]
```diff
@@ -38,9 +39,10 @@ jobs:
...
```
Comment (oscard0m) [inline .github/workflows/CI.yml:45]:
```suggestion
          cache: yarn
```

File: .github/workflows/Deploy.yml [hunk 3]
```diff
@@ -14,10 +14,11 @@ jobs:
 
     steps:
       - uses: actions/checkout@v2
-      - uses: actions/setup-node@v1
+      - uses: actions/setup-node@v2
         with:
           node-version: "12.x"
           registry-url: "https://registry.npmjs.org"
+          cache: npm
```
Comment (oscard0m) [inline .github/workflows/Deploy.yml:21]:
```suggestion
          cache: yarn
```

File: .github/workflows/DeployExtensionsProd.yml [hunk 4]
```diff
@@ -11,10 +11,11 @@ jobs:
 
     steps:
       - uses: actions/checkout@v2
-      - uses: actions/setup-node@v1
+      - uses: actions/setup-node@v2
         with:
           node-version: "12.x"
           registry-url: "https://registry.npmjs.org"
+          cache: npm
```
Comment (oscard0m) [inline .github/workflows/DeployExtensionsProd.yml:18]:
```suggestion
          cache: yarn
```

File: .github/workflows/DeploySvelte2tsxProd.yml [hunk 5]
```diff
@@ -11,10 +11,11 @@ jobs:
 
     steps:
       - uses: actions/checkout@v2
-      - uses: actions/setup-node@v1
+      - uses: actions/setup-node@v2
         with:
           node-version: "12.x"
           registry-url: "https://registry.npmjs.org"
+          cache: npm
```
Comment (oscard0m) [inline .github/workflows/DeploySvelte2tsxProd.yml:18]:
```suggestion
          cache: yarn
```

File: .github/workflows/DeploySvelteCheckProd.yml [hunk 6]
```diff
@@ -11,10 +11,11 @@ jobs:
 
     steps:
       - uses: actions/checkout@v2
-      - uses: actions/setup-node@v1
+      - uses: actions/setup-node@v2
         with:
           node-version: "12.x"
           registry-url: "https://registry.npmjs.org"
+          cache: npm
```
Comment (oscard0m) [inline .github/workflows/DeploySvelteCheckProd.yml:18]:
```suggestion
          cache: yarn
```

File: .github/workflows/DeploySvelteLanguageServerProd.yml [hunk 7]
```diff
@@ -11,10 +11,11 @@ jobs:
 
     steps:
       - uses: actions/checkout@v2
-      - uses: actions/setup-node@v1
+      - uses: actions/setup-node@v2
         with:
           node-version: "12.x"
           registry-url: "https://registry.npmjs.org"
+          cache: npm
```
Comment (oscard0m) [inline .github/workflows/DeploySvelteLanguageServerProd.yml:18]:
```suggestion
          cache: yarn
```

File: .github/workflows/DeployTypescriptPluginProd.yaml [hunk 8]
```diff
@@ -11,10 +11,11 @@ jobs:
 
     steps:
       - uses: actions/checkout@v2
-      - uses: actions/setup-node@v1
+      - uses: actions/setup-node@v2
         with:
           node-version: "12.x"
           registry-url: "https://registry.npmjs.org"
+          cache: npm
```
Comment (oscard0m) [inline .github/workflows/DeployTypescriptPluginProd.yaml:18]:
```suggestion
          cache: yarn
```

File: .github/workflows/CI.yml [hunk 9]
```diff
@@ -38,9 +39,10 @@ jobs:
 
     steps:
       - uses: actions/checkout@v1
-      - uses: actions/setup-node@v1
+      - uses: actions/setup-node@v2
         with:
           node-version: "12.x"
+          cache: yarn
```
Comment (dummdidumm) [inline .github/workflows/CI.yml:45]: There's already some caching logic in this file. I guess the old logic can be removed?

Comment (oscard0m) [reply .github/workflows/CI.yml:33]: Yep, I think it can be removed.

**Pull request merged**